### ISSUE TITLE
SlevomatCodingStandard.PHP.UselessParentheses.UselessParentheses - false positive with modulo operator

### ISSUE THREAD

**Issue by rvock** [slevomat/coding-standard#1678]
```php
<?php
declare(strict_types = 1);

var_dump(!(12 % 12));
var_dump(!12 % 12);
```

Output:
```
bool(true)
int(0)
```

Here I get an error, that the first parentheses are useless. But that's not the case, because the output is different, if I remove them.

The `!` operator has a higher precedence than the module `%` operator:
https://www.php.net/manual/en/language.operators.precedence.php

**Comment by bkdotcom** [comment]
See also #1674, #1672, & #1678

This sniff doesn't seem to take context into account  (only looks within, but not outward)

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/2ed413255590b2b2414d0edfbcba5280a6d536f4
